synapse/handlers/directory.py

```diff
@@ -133,12 +138,12 @@ async def create_association(
                     )
 
             if not await self.spam_checker.user_may_create_room_alias(
-                user_id, room_alias
+                user_id, room_alias_str
```

Comment:
This was a real bug as far as I can tell. I think it has always existed?

Reply:
Hum, I wonder if any spam checkers in the wild assume its a `RoomAlias`?

Reply:
I wasn't sure what to do here, it's possible of course.

Mjolnir doesn't use this method: https://github.com/matrix-org/mjolnir/blob/master/synapse_antispam/mjolnir/antispam.py#L121-L122 (although there's a PR for a rewrite which does: https://github.com/matrix-org/mjolnir/pull/78/files#diff-abf5313147c56e3bf705f659a3b8535242f30744087be63343ee8281d43ded2bR131-R142).

🤷

Reply:
I'm hesitant to change behaviour here, in case it *is* used. I wonder if the less risky thing to do is to change the docs to say that its a string?

Reply:
> I wonder if the less risky thing to do is to change the docs to say that its a string?

Do you mean to change the docs to say it is a `RoomAlias`?

The [docs](https://github.com/matrix-org/synapse/blob/develop/docs/spam_checker.md#python-spam-checker-class) don't really say anything, it would be updating the type hint on the spam checker, I think.

Reply:
Yes, sorry, I got them the wrong way round. If the docs don't say anything then I think we should just update the type hint. Either someone is using it already and changing it will explode, or no one is using it and there's no harm changing it?

Reply:
That makes sense to me. 👍